UT4MasterServer/Services/RatingsService.cs

```diff
@@ -0,0 +1,260 @@
+﻿using MongoDB.Driver;
+using Newtonsoft.Json.Linq;
+using UT4MasterServer.Models;
+using UT4MasterServer.Models.Requests;
+using UT4MasterServer.Other;
+
+namespace UT4MasterServer.Services;
+
+public sealed class RatingsService
+{
+	private readonly IMongoCollection<Rating> ratingsCollection;
+
+	public RatingsService(DatabaseContext dbContext)
+	{
+		ratingsCollection = dbContext.Database.GetCollection<Rating>("ratings");
+	}
+
+	public async Task<MMRBulk> GetRatingsAsync(EpicID accountID, MMRBulk mmrBulk)
+	{
+		var filter = Builders<Rating>.Filter.Eq(f => f.AccountID, accountID);
+		var ratings = await ratingsCollection.Find(filter).FirstOrDefaultAsync();
+
+		var result = new MMRBulk();
+
+		foreach (var ratingType in mmrBulk.RatingTypes)
+		{
+
+			switch (ratingType)
+			{
+				case "SkillRating":
+					result.RatingTypes.Add(ratingType);
+					result.Ratings.Add(ratings.SkillRating / Rating.Precision);
+					result.NumGamesPlayed.Add(ratings.SkillRatingGamesPlayed);
+					break;
+
+				case "TDMSkillRating":
+					result.RatingTypes.Add(ratingType);
+					result.Ratings.Add(ratings.TDMSkillRating / Rating.Precision);
+					result.NumGamesPlayed.Add(ratings.TDMSkillRatingGamesPlayed);
+					break;
+
+				case "CTFSkillRating":
+					result.RatingTypes.Add(ratingType);
+					result.Ratings.Add(ratings.CTFSkillRating / Rating.Precision);
+					result.NumGamesPlayed.Add(ratings.CTFSkillRatingGamesPlayed);
+					break;
+
+				case "ShowdownSkillRating":
+					result.RatingTypes.Add(ratingType);
+					result.Ratings.Add(ratings.ShowdownSkillRating / Rating.Precision);
+					result.NumGamesPlayed.Add(ratings.ShowdownSkillRatingGamesPlayed);
+					break;
+
+				case "FlagRunSkillRating":
+					result.RatingTypes.Add(ratingType);
+					result.Ratings.Add(ratings.FlagRunSkillRating / Rating.Precision);
+					result.NumGamesPlayed.Add(ratings.FlagRunSkillRatingGamesPlayed);
+					break;
+
+				case "DMSkillRating":
+					result.RatingTypes.Add(ratingType);
+					result.Ratings.Add(ratings.DMSkillRating / Rating.Precision);
+					result.NumGamesPlayed.Add(ratings.DMSkillRatingGamesPlayed);
+					break;
+			}
+		}
+
+		return result;
+	}
+
+	public async Task<JObject> GetRatingAsync(EpicID accountID, string ratingType)
```

Comment:
This is just a preference comment, you don't have to do this if you don't feel like it.

We are trying to go away from Newtonsoft (#33) so creating a class for the returned object would be great.

Reply:
I will remove JObject, I just followed the existing code.